docs/10-lifecycle.md

```diff
@@ -0,0 +1,81 @@
+# Lifecycle
+
+ContainerPilot's core functionality is centered around the concepts of jobs, the events that trigger those jobs or that jobs emit, and watches for the external service discovery catalog (Consul).
```

Comment:
What about starting this section by introducing the importance of Lifecycles themselves within ContainerPilot? This paragraph seems to introduce Jobs which are brought in as a secondary concept under the next paragraph.